pkg/platform/runtime2/api/api.go

```diff
@@ -0,0 +1,29 @@
+package api
```

Comment:
Well, it actually is more of  a `model` in our current terminology.  Should it live here though, or in `pkg/platform/model`?

Reply:
`model` is for structuring requests and processing responses, it shouldn't deal with api configuration or actually processing the requests/responses on the network level.

I'd say most of what I'm seeing here should live in the existing `api` directory. In fact, aren't we already covering this there?

Reply:
Yes, `api` was an unfortunate name for this probably.  I never intended the `ClientProvider` to actually deal with requests on the network level.  In its default installation, it would just be a thin wrapper around all the APIs that we need to use here: `graphql`, `headchef` and `buildlogstream`.  But as it is three of them, I find this concept warranted.  Its main purpose is to facilitate the `client.Mock` implementation that fulfills the `ClientProvider` interface without doing any network requests (not even to a mocked http server).

In my opinion this is more of a `model` therefore, as in the implementations we (can) do some post-processing of API calls.